### PR TITLE
Duplicate keys

### PR DESCRIPTION
**Pull request by phlax**
handle duplicate keys

### CODE REVIEW

File: translate/storage/pypo.py [hunk 1]
```diff
@@ -791,7 +795,13 @@ def addcomment(thepo):
                         origpo.msgctxt.append('"%s"' % escapeforpo(" ".join(origpo.getlocations())))
                         markedpos.append(thepo)
                     thepo.msgctxt.append('"%s"' % escapeforpo(" ".join(thepo.getlocations())))
-                    uniqueunits.append(thepo)
+                    if not thepo.msgctxt == id_dict[id].msgctxt:
```
Comment (phlax) [inline translate/storage/pypo.py]: this condition could be improved

File: translate/storage/pypo.py [hunk 2]
```diff
@@ -791,7 +795,13 @@ def addcomment(thepo):
...
+                        uniqueunits.append(thepo)
+                    else:
+                        logger.warn(
+                            "Duplicate unit found with msgctx of '%s' and source '%s'",
```
Comment (phlax) [inline translate/storage/pypo.py]: this should be msgctxt

Comment (dwaynebailey) [inline translate/storage/pypo.py]: How about `"Duplicate unit found with msgctxt of '%s' and msgid '%s'"`

**Comment by unho** [comment]
What about cpo implementation?

**Comment by phlax** [comment]
> What about cpo implementation?

added

**Comment by unho** [comment]
Also needs to be added to fpo implementation.

**Comment by phlax** [comment]
closed in favour of #3587

**Pull request closed without merging**